### PR TITLE
update matrix nms op to api 2.0

### PR DESCRIPTION
**Pull request by wangxinxin08**
### PR types 
Function optimization

### PR changes
OPs

### Describe
Update matrix nms to be suitable for dygraph. Add RoisNum for output  which is dispensable and compatible with inference of previous Paddle version. The new output is only used in dygraph detection models. 
The test result on ppyolo shows that the latest paddle version could use the inference model from previous paddle version correctly.

预测兼容性测试：

测试方法：
在PaddleDetection中使用 [export_model.py](https://github.com/PaddlePaddle/PaddleDetection/blob/release/0.4/tools/export_model.py)  导出模型，使用[deploy/python/infer.py](https://github.com/PaddlePaddle/PaddleDetection/blob/release/0.4/deploy/python/infer.py)进行预测，对比检测结果。测试模型为[ppyolo](https://github.com/PaddlePaddle/PaddleDetection/blob/release/0.4/configs/ppyolo/ppyolo.yml) ，其中包含了改动的OP。

case1：
使用paddle1.8.4版本保存模型，在1.8.4版本和develop版本分别预测
1.8.4版本预测结果：
image
develop版本预测结果：
image

case2:
使用develop版本保存模型，在1.8.4版本和develop版本分别预测
1.8.4版本预测结果：
image
develop版本预测结果：
image
测试结论：
新增输出是兼容老版本的

### CODE REVIEW

File: paddle/fluid/operators/detection/matrix_nms_op.cc [hunk 1]
```diff
@@ -369,7 +384,9 @@ This operator support multi-class and batched inputs. It applying NMS
 independently for each class. The outputs is a 2-D LoDTenosr, for each
 image, the offsets in first dimension of LoDTensor are called LoD, the number
 of offset is N + 1, where N is the batch size. If LoD[i + 1] - LoD[i] == 0,
-means there is no detected bbox for this image.
+means there is no detected bbox for this image. Now this operator has one more
+ouput, which is RoisNum. The size if RoisNum is N, RoisNum[i] means the number of
```
Comment (jerrywgz) [inline paddle/fluid/operators/detection/matrix_nms_op.cc]: The size of RoisNum

**Review by jerrywgz — COMMENTED** [review]
need to add op_outs_map in paddle/fluid/pybind/op_function_generator.cc

**Comment by jerrywgz** [comment]
compatibility check for inference is needed as well.

File: paddle/fluid/operators/detection/matrix_nms_op.cc (hunk 1 above)
Comment (wangxinxin08) [reply paddle/fluid/operators/detection/matrix_nms_op.cc]: Resolved, thanks!

**Review by zhiqiu — APPROVED** [review]
LGTM for op_function_generator.cc

**Pull request merged**